corehq/apps/app_manager/util.py

```diff
@@ -409,7 +409,7 @@ def get_casedb_schema(form):
     This lists all case types and their properties for the given app.
     """
     app = form.get_app()
-    base_case_type = form.get_module().case_type
+    base_case_type = form.get_module().case_type if form.requires_case() else None
```

Comment:
Maybe I just need to know more about that. How is it kept up to date? Is it recalculated every time an app is saved?